subprojects/core/src/main/java/org/gradle/internal/scripts/BuildScriptCompilationAndInstrumentation.java

```diff
@@ -61,6 +67,15 @@ public BuildScriptCompilationAndInstrumentation(
         this.transformFactory = transformFactory;
     }
 
+    @Override
+    public Optional<CachingDisabledReason> shouldDisableCaching(@Nullable OverlappingOutputs detectedOverlappingOutputs) {
+        if (System.getProperty(CACHING_DISABLED_PROPERTY) != null) {
```

Comment:
❓ Also here, should we also check if value is true or false?

Reply:
The plan is to support usage like `gradle -Dorg.gradle.experimental.script-caching-disabled foo` (without requiring a value).

For transforms, I've updated this to allow a comma-separated list of transforms to disbaled, with empty string meaning disable all transforms. I suggest we keep this logic for scripts.

Since this is an experimental property, I'm not overly concerned with usability.